src/molecules/Amount/Amount.native.js

```diff
@@ -0,0 +1,169 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { Text as NativeText, Platform } from 'react-native';
+import styled from 'styled-components/native';
+import automation from '../../_helpers/automation-attributes';
+import Flex from '../../atoms/Flex';
+import Text from '../../atoms/Text';
+import Heading from '../../atoms/Heading';
+
+const styles = {
+  text: {
+    color({ size }) {
+      switch (size) {
+        case 'medium':
+          return 'shade.940';
+        case 'large':
+          return 'shade.940';
+        case 'xlarge':
+          return 'shade.940';
+        case 'xxlarge':
+          return 'shade.940';
+        case 'xxxlarge':
+          return 'shade.950';
+        default:
+          return 'shade.940';
+      }
+    },
+    size({ size }) {
+      switch (size) {
+        case 'medium':
+          return 'xxsmall';
+        case 'large':
+          return 'xsmall';
+        case 'xlarge':
+          return 'medium';
+        case 'xxlarge':
+          return 'medium';
+        case 'xxxlarge':
+          return 'large';
+        default:
+          return 'xxsmall';
+      }
+    },
+    weight({ size }) {
+      switch (size) {
+        case 'medium':
+          return 'regular';
+        case 'large':
+          return 'bold';
+        case 'xlarge':
+          return 'bold';
+        case 'xxlarge':
+          return 'bold';
+        case 'xxxlarge':
+          return 'bold';
+        default:
+          return 'bold';
+      }
+    },
+    lineHeight({ size }) {
+      switch (size) {
+        case 'medium':
+          return 'medium';
+        case 'large':
+          return 'medium';
+        case 'xlarge':
+          return 'xlarge';
+        case 'xxlarge':
+          return 'xxlarge';
+        case 'xxxlarge':
+          return 'xxlarge';
+        default:
+          return 'xxlarge';
+      }
+    },
+  },
+  baseText: {
+    alignSelf({ align }) {
+      switch (align) {
+        case 'left':
+          return 'flex-start';
+        case 'center':
+          return 'center';
+        case 'right':
+          return 'flex-end';
+        default:
+          return 'flex-start';
+      }
+    },
+  },
+};
+
+const BaseText = styled(NativeText)`
+  align-self: ${styles.baseText.alignSelf};
+`;
+
+const getFractionalDisplay = (fractionalPart = '') => {
+  switch (fractionalPart.length) {
+    case 0:
+      return '00';
+    case 1:
+      return `${fractionalPart}0`;
+    case 2:
+      return fractionalPart;
+    default:
+      return fractionalPart.slice(0, 2);
+  }
+};
+
+const getLocaleString = (numberString) => {
+  if (Platform.OS === 'android') {
+    // only android needs this polyfill
+    require('intl');
+    require('intl/locale-data/jsonp/en-IN');
+    return new Intl.NumberFormat('en-IN').format(numberString);
+  } else {
+    return parseInt(numberString, 10).toLocaleString('en-IN');
```

Comment:
```js
const formatAmount = ({ amount, currency }) => {
  //polyfill here

  //normal toLocale method
  const formattedAmount = (+amount).toLocaleString("en-IN", {
    style: "currency",
    currency,
    minimumFractionDigits: 2,
    maximumFractionDigits: 2
  });
  return formattedAmount;
};
```

Reply:
You can then do splits and other things to apply stylings to individual pieces below